sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -1653,7 +1653,7 @@ export interface SearchIndexerWarning {
 
 // @public
 export class SearchIndexingBufferedSender<T> {
-    constructor(client: IndexDocumentsClient<T>, options?: SearchIndexingBufferedSenderOptions);
+    constructor(client: IndexDocumentsClient<T>, documentKeyRetriever: (document: T) => string, options?: SearchIndexingBufferedSenderOptions);
```

Comment:
Let's tighten the typing for `documentKeyRetriever`: 

```ts
interface documentKeyRetriever<T> {
    (document: T): keyof T;
}
```

This lets us assert that the returned string is actually a key of the document itself:

```ts
interface Thing {
    a: string;
    b: number;
}

const foo: documentKeyRetriever<Thing> = (thing: Thing) => {
    // returning "c" would be an error here
    return "a";
}
```

Reply:
I agree that the new return type ```keyof T``` will make it strict. But, there is no requirement that the ```key``` of the document should be a part of it. The user can add whatever logic he/she wants to uniquely identify the document. The return type has to be a string. That is the only requirement.

Reply:
Also, we are no getting the ```key```. We are getting the ```value```. So, in the provide example above, the return value of method is not "a"/"b". It is the value of that "a"/"b". Am I making sense?

Reply:
Oh right! And keys have to be strings, so the typing is already as correct as it needs to be. Yeah ignore this one.